src/server/protocol.ts

```diff
@@ -546,7 +546,15 @@ namespace ts.server.protocol {
         command: CommandTypes.GetApplicableRefactors;
         arguments: GetApplicableRefactorsRequestArgs;
     }
-    export type GetApplicableRefactorsRequestArgs = FileLocationOrRangeRequestArgs;
+    export type GetApplicableRefactorsRequestArgs = FileLocationOrRangeRequestArgs & {
+        triggerReason?: RefactorTriggerReason;
+    };
+
+    export type RefactorTriggerReason = RefactorInvokedReason;
+
+    export interface RefactorInvokedReason {
+        kind: "invoked";

```

Comment:
Seems like there should be more than one possible kind?

Reply:
Or maybe we're treating `undefined` as a default kind that means "implicitly invoked"?

Reply:
>Or maybe we're treating undefined as a default kind that means "implicitly invoked"?

Yes, that was the idea. It probably makes more sense to have take an explicit `implicit`  kind and treat it the same as an `undefined` `RefactorTriggerReason`.

```ts
export enum RefactorTriggerReason {
    Implicit = "implicit",
    Invoked = "invoked",
}
```